### PR TITLE
added spack_config and spack_package data

### PR DESCRIPTION
**Pull request by utkarshgupta95**
resolves ACCESS-NRI/experiment-metadb#15

### CODE REVIEW

**Comment by CodeGat** [comment]
Thanks @utkarshgupta95 - I might add some commits as well so the workflow works

**Review by CodeGat — APPROVED** [review]
The stuff that @utkarshgupta95 has done looks good. I'll need someone to verify that my stuff is good, as well.

File: scripts/generate-build-metadata.bash [hunk 1]
```diff
@@ -32,13 +36,17 @@ model=$(jq \
...
+  --arg spack_packages_version "$spack_packages_version" \
+  --arg spack_config_version "$spack_config_version" \
   --argjson spack "$spack" \
   '.concrete_specs | to_entries[] | select(.value.name == $model)
   | {
       spack_hash: .key,
       spec: (.value.name + "@" + .value.version),
       created_at: $release_time,
       release_url: $release_url,
+      spack_package: $spack_packages_version,
```
Comment (aidanheerdegen) [inline scripts/generate-build-metadata.bash:48]:
```suggestion
      spack_packages: $spack_packages_version,
```
It's a small detail, but we should be consistent with the naming of `spack_packages`

File: tests/scripts/generate-build-metadata/valid/build_metadata.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
     "spec": "access-om2@git.2024.03.0=2024.03.0",
     "created_at": "2024-03-27T05:29:36Z",
     "release_url": "http://example.org/releases",
+    "spack_package": "2023.11.23",
```
Comment (aidanheerdegen) [inline tests/scripts/generate-build-metadata/valid/build_metadata.json:7]:
```suggestion
    "spack_packages": "2023.11.23",
```

File: tools/release_provenance/models.py [hunk 3]
```diff
@@ -33,6 +33,8 @@ class ModelBuild(Base):
     spack_hash = Column(String, primary_key=True, index=True)
     spec = Column(String, nullable=False)
     spack_version = Column(String, ForeignKey("spack_version.commit"))
+    spack_package = Column(String)
```
Comment (aidanheerdegen) [inline tools/release_provenance/models.py:36]:
```suggestion
    spack_packages = Column(String)
```

File: tools/release_provenance/save_release.py [hunk 4]
```diff
@@ -33,6 +33,8 @@ def get_model_build(model_build_data):
     if model_build == None:
         model_build = ModelBuild()
         model_build.spack_version = get_spack_version(model_build_data["spack_version"])
+        model_build.spack_config = model_build_data["spack_config"]
+        model_build.spack_package = model_build_data["spack_package"]
```
Comment (aidanheerdegen) [inline tools/release_provenance/save_release.py:37]:
```suggestion
        model_build.spack_packages = model_build_data["spack_packages"]
```

File: tools/release_provenance/test_release_data.json [hunk 5]
```diff
@@ -12,10 +12,12 @@
       }
     ],
     "model_build": {
-      "spack_hash": "f4f2qe5b3c22q4dtzzqs5o2iygy435rh",
+      "spack_hash": "f4f2qe5b3c22q4dtzzqs5o2iygy435r",
       "spec": "access-om2@git.2023.11.23=2023.11.23",
       "created_at": "2024/02/13 11:49:00",
       "release_url": "https://github.com/ACCESS-NRI/ACCESS-OM3/releases",
+      "spack_package": "0.20.3",
```
Comment (aidanheerdegen) [inline tools/release_provenance/test_release_data.json:19]:
```suggestion
      "spack_packages": "0.20.3",
```

**Review by aidanheerdegen — CHANGES_REQUESTED** [review]
Just a minor quibble about naming

**Comment by utkarshgupta95** [comment]
@aidanheerdegen just to confirm `spack_packages` is not an array?

**Review by aidanheerdegen — APPROVED** [review]
Looks great, thanks!

**Pull request merged**